cmd/cdi/clients/github_test.go

```diff
@@ -0,0 +1,101 @@
+package clients_test
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/joesustaric/cdi/cmd/cdi/clients"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewGitHubClient_Error_When_No_Token(t *testing.T) {
+	_, e := clients.NewGitHubClient(nil)
+
+	assert.NotNil(t, e)
+}
+
+func TestNewGitHubClient_No_Error_With_Token(t *testing.T) {
+	os.Setenv("GITHUB_TOKEN", "123456")
+	defer os.Unsetenv("GITHUB_TOKEN")
+
+	client, e := clients.NewGitHubClient(nil)
+
+	assert.Nil(t, e)
+	assert.NotNil(t, client)
+}
+
+func TestRawBranchClount_Returns_Count_Of_Branches(t *testing.T) {
+	os.Setenv("GITHUB_TOKEN", "123456")
+	defer os.Unsetenv("GITHUB_TOKEN")
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/graphql", func(w http.ResponseWriter, req *http.Request) {
+
+		got := mustRead(req.Body)
+		want := `{"query":"query($organisationName:String!$repositoryName:String!){repository(owner: $organisationName, name: $repositoryName){id,refs(refPrefix: \"refs/heads/\"){totalCount}}}","variables":{"organisationName":"some-org","repositoryName":"some-repo"}}` + "\n"
```

Comment:
Not sure what the best golang way is to format a multi line string. I like them splatted out so it's easier to read. Every time I tried it it would pick up the white space and the assertions would blow up.